src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Text.cs

```diff
@@ -143,41 +145,45 @@ public override bool CheckTypes(CheckTypesContext checkTypesContext, TexlNode[]
 
         internal static void ValidateFormatArgs(string name, CheckTypesContext checkTypesContext, TexlNode[] args, DType[] argTypes, IErrorContainer errors, ref Dictionary<TexlNode, DType> nodeToCoercedTypeMap, ref bool isValid)
         {
-            if (BuiltInEnums.DateTimeFormatEnum.FormulaType._type.Accepts(argTypes[1], exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: checkTypesContext.Features.PowerFxV1CompatibilityRules))
+            bool hasConstFormat = BinderUtils.TryGetConstantValue(checkTypesContext, args[1], out var formatArg);
+            if (checkTypesContext.Features.PowerFxV1CompatibilityRules && argTypes[0] != DType.UntypedObject && !TextFormatUtils.AllowedListToUseFormatString.Contains(argTypes[0]))
             {
-                // Coerce enum values to string
-                CollectionUtils.Add(ref nodeToCoercedTypeMap, args[1], DType.String);
-            }
-            else if (!DType.String.Accepts(argTypes[1], exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: checkTypesContext.Features.PowerFxV1CompatibilityRules))
-            {
-                errors.EnsureError(DocumentErrorSeverity.Severe, args[1], TexlStrings.ErrStringExpected);
-                isValid = false;
+                if (!hasConstFormat || !string.IsNullOrEmpty(formatArg))
```

Comment:
>if (!hasConstFormat || !string.IsNullOrEmpty(formatArg)) [](http://example.com/codeflow?start=16&length=56)

In this case, we don't need to check the format. Text can only be called with a first argument of type text / Boolean if there is a single argument (i.e., as a pure conversion function). Something such as Text(aTextValue, "") should be compile-time error as well.

Reply:
Updated here